firmware/README.md

```diff
@@ -4,18 +4,26 @@ CYW43xx WiFi SoC firmware
 This directory contains firmware patch blobs that need to be downloaded on to the
 CYW43xx SoC in order for it to function correctly.
 
-The firmware is padded to 512 bytes and then the CLM appended to that, to create the
-combined binary file.
+The Wifi firmware and CLM data is padded to 512 bytes. Then the Wifi firmware, CLM data and BT firmware are all appended together,
+to create the combined binary file.
 
 For example:
 
     $ cp 43439A0.bin 43439A0_padded.bin
     $ dd if=/dev/zero of=43439A0_padded.bin bs=1 count=1 seek=$(( ($(stat -c %s 43439A0.bin) / 512) * 512 + 512 - 1))
     $ cat 43439A0_padded.bin 43439A0.clm_blob > 43439A0-7.95.49.00.combined
 
-Note that the number of dots in the filename is significant, due to the way symbols
-are renamed by the linker when it is included in a compiled application.
+This binary is converted to a header file, e.g. xxd -i 43439A0-7.95.49.00.combined
+The macros `CYW43_WIFI_FW_LEN`, `CYW43_CLM_LEN` specify the unpadded size of the original firmware binaries in bytes.
 
-When updating this firmware, check and update the values of `CYW43_FW_LEN` and
-`CYW43_CLM_LEN`.  These should be the original, unpadded size of the two binaries
-in bytes.  The array size should be the size of the combined file.
+The BT firmware is supplied as a source file containing a static array with ascii hex data. We process this to reduce it in size.
+See CYW43_USE_HEX_BTFW in the SDK to load the original source file. The BT fw is now formatted like this...
+
+1 byte: number of characters in version string including null terminator
+n bytes: zero terminated version string
+1 byte: number of records
+...
```

Comment:
I guess these dots mean "each record has the following form"?  If so would be good to write that explicitly.